rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+This version would allow users to choose selective privacy, deciding whether they want their transactions to be public, private, shielded, or deshielded. 
+
+For example, users engaging in sensitive financial transactions might opt for full privacy, while users making routine, low-stakes transactions might prefer to keep them public 
+for simplicity and efficiency. NSSA enables this flexibility, even though the Dapp itself originally did not support privacy features.
+
+**Key Advantage: Decoupling from the Host Chain**
+
+A key feature of NSSA is that it operates independently of the privacy characteristics of the host blockchain. Whether the host chain is fully transparent or fully private, 
+the Nescience state-separation architecture can be deployed on top of it, offering users the ability to choose their own privacy settings. 
+This decoupling from the host chain’s inherent privacy model is critical, as it allows users to benefit from selective privacy even in environments that were not originally designed to offer it.
+
+In **fully private chains**, NSSA allows users to selectively reveal transaction details when compliance with regulations or other requirements is necessary. 
+In **fully transparent chains**, NSSA allows users to maintain privacy for specific transactions, offering flexibility that would not otherwise be possible.
+
+**Conclusion**
+
+NSSA provides a powerful tool for Dapp developers who want to offer **selective privacy** to their users without the need for a complete overhaul of their existing systems. 
+By deploying the Nescience state-separation architecture as an add-on, Dapps can give users the ability to choose their own privacy settings, whether they are operating on 
+transparent or private blockchains. This flexibility makes NSSA a valuable option for any Dapp looking to provide enhanced privacy options while maintaining efficiency and ease of use.
+
+# B. Design
+In this section, we will delve into the core design components of the Nescience State Separation Architecture, covering its key structural elements and the mechanisms 
+that drive its functionality. We will explore the following topics:
+
+1. **Architecture's Components**: An in-depth look at the foundational building blocks of NSSA, including the public and private states, UTXO structures, zkVM, and smart contracts. 
+These components work together to facilitate secure, flexible, and scalable transactions within the architecture.
+
+2. **General Execution Overview**: We will outline the overall flow of transaction execution within NSSA, describing how users interact with the system and how the architecture 
+supports various types of executions—public, private, shielded, and deshielded—while preserving privacy and efficiency.
+
+3. **Execution Processes and UTXO Management**: This section will focus on the lifecycle of UTXOs within the architecture, from their generation to consumption. 
```

Comment:
Replace:
Execution Processes and UTXO Management

With:
Execution processes and UTXO management